californium-core/src/main/java/org/eclipse/californium/core/observe/ObserveRelation.java

```diff
@@ -163,7 +163,7 @@ public Exchange getExchange() {
 	 *
 	 * @return the source address
 	 */
-	public InetSocketAddress getSource() {
+	public final InetSocketAddress getSource() {
```

Comment:
This is same as point 1 so lets park it for now until we have more structured solution.